### PR TITLE
Base DDD implementation

### PR DESCRIPTION
**Pull request by williamabreu**
#### 📝 Description

This PR implements the base DDD model to use in other projects. Also, configures a TypeScript boilerplate to follow in new projects.

#### 🛠️ Changes

- Added banner as a remainder to manual processes on pull request template.
- Added VS Code config<sup>1</sup>.
- Added makefile to ease the run of processual tasks.
- Configured a TypeScript boilerplate (babel, eslint, prettier, jest, etc).
- Added DDD models, messages and exceptions base classes.
- Added unit tests for the base classes as proof of concept and usage example.

#### ⚠️ Notes

1. The VS Code config is platform-independent, keeping it on track may ease the setup of a new environment (like if a new developer joins the team).

### CODE REVIEW

File: src/domain/models/Entity.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import objectHash from 'object-hash';
+import EntityDiscardedError from '../exceptions/EntityDiscardedError';
+import Props from '../../interfaces/Props';
+
+/**
+ * Represent the base element in the domain model, for entities and its aggregates.
+ */
+export default abstract class Entity<Id_T> {
+  private _id: Id_T;
```
Comment (andreyrcdias) [inline src/domain/models/Entity.ts:17]: Is it necessary to use `_` before the attr even being declared as `private`? Sorry, do not know much about TypeScript.

**Review by andreyrcdias — APPROVED** [review]
@williamabreu really good job!

Nothing to point out about this implementation.

🚀😤

File: src/domain/models/Entity.ts (hunk 1 above)
Comment (williamabreu) [reply src/domain/models/Entity.ts:17]: Not really, but as I'm using the get/set properties (like python does), I need to distinguish the field itself from the property in this case. The best practice is to use an underscore prefix.

Comment (andreyrcdias) [reply src/domain/models/Entity.ts:17]: Ok, i got it

**Pull request merged**